### PR TITLE
fix: added migrate-legacy-policies guide closes #634

### LINKED ISSUE #634: docs: write a guide to migrate from v0.5 policies and describe v0.6 permissions benefits
From February slack harvest

> can anyone tell me what’s the benefits of the new implementation of keto’s permissions in v0.6 comparing to v0.5 ? reading the documentation makes me feel v0.5 is simpler to understand and more flexible.

It can be useful for v0.5 users to upgrade their permissions models

### PR DESCRIPTION
**Pull request by gen1us2k**
## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [x] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

**Comment by aeneasr** [comment]
@zepatrik PTAL

File: docs/keto/guides/migrating-legacy-policies.mdx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    "blog_posts:3"
+  ],
+  "actions": ["delete", "create", "read", "modify"],
+  "effect": "allow"
+}
+```
+
+## Rewriting it to relation tuples
+
+According to the example above we need to create required [namespace](https://www.ory.sh/docs/keto/concepts/namespaces) and [relationship tuple](https://www.ory.sh/docs/keto/concepts/relation-tuples)
```
Comment (zepatrik) [inline docs/keto/guides/migrating-legacy-policies.mdx:27]: Just a tiny naming thing, but we use relation tuple everywhere, not relationship.
```suggestion
According to the example above we need to create required [namespace](https://www.ory.sh/docs/keto/concepts/namespaces) and [relation tuple](https://www.ory.sh/docs/keto/concepts/relation-tuples)
```

File: docs/keto/guides/migrating-legacy-policies.mdx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+General mapping from old to new policies
+
+- Subjects -> Subject/Subject Sets
```
Comment (zepatrik) [inline docs/keto/guides/migrating-legacy-policies.mdx:31]:
```suggestion
- Subjects -> Subject IDs or Subject Sets
```

File: docs/keto/guides/migrating-legacy-policies.mdx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+- Resources -> Namespaces/Objects
```
Comment (zepatrik) [inline docs/keto/guides/migrating-legacy-policies.mdx:32]:
```suggestion
- Resources -> Objects scoped by namespaces
```

File: docs/keto/guides/migrating-legacy-policies.mdx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+- Actions -> Relations
+- Effect -> Became obsolete or can be considered as Relations
+
+We need to have `blog_posts` namespace for our example. Let's add the following content to `keto.yml` configuration file. You can find a good template [here](https://github.com/ory/keto/blob/master/config/keto.yml).
```
Comment (zepatrik) [inline docs/keto/guides/migrating-legacy-policies.mdx:36]: Don't use links to master but always permalinks.
```suggestion
We need to have `blog_posts` namespace for our example. Let's add the following content to `keto.yml` configuration file. You can find a good template [here](https://github.com/ory/keto/blob/3453d47e67cfe45ad1389a2aa25fd65dfdeb114d/config/keto.yml).
```

Maybe rather link to the config reference? https://www.ory.sh/docs/keto/reference/configuration

File: docs/keto/guides/migrating-legacy-policies.mdx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+
+```yaml
+namespaces:
+  - id: 0
+  name: blog_posts
```
Comment (zepatrik) [inline docs/keto/guides/migrating-legacy-policies.mdx:42]:
```suggestion
namespaces:
  - id: 0
    name: blog_posts
```

File: docs/keto/guides/migrating-legacy-policies.mdx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+serve:
+  read:
+    host: 0.0.0.0
+    port: 4466
+  write:
+    host: 0.0.0.0
+    port: 4467
+```
+
+## Alice relationship tuples
```
Comment (zepatrik) [inline docs/keto/guides/migrating-legacy-policies.mdx:53]:
```suggestion
## Alice Relation Tuples
```

File: docs/keto/guides/migrating-legacy-policies.mdx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+blog_posts:2#modify@bob
+blog_posts:2#delete@bob
+blog_posts:2#create@bob
+blog_posts:3#read@bob
+blog_posts:3#modify@bob
+blog_posts:3#delete@bob
+blog_posts:3#create@bob
+```
+
+## Creating rules using Cli
```
Comment (zepatrik) [inline docs/keto/guides/migrating-legacy-policies.mdx:88]:
```suggestion
## Creating Relation Tuples using the CLI
```

File: docs/keto/guides/migrating-legacy-policies.mdx [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+This example uses keto cli to create the relation tuple using [write API](../concepts/api-overview#write-apis)
```
Comment (zepatrik) [inline docs/keto/guides/migrating-legacy-policies.mdx:90]:
```suggestion
This example uses the Ory Keto CLI to create the relation tuple using the [write API](../concepts/api-overview#write-apis)
```

File: docs/keto/guides/migrating-legacy-policies.mdx [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+
+
+```bash
+keto relation-tuple parse alice_policies- --format json | \
```
Comment (zepatrik) [inline docs/keto/guides/migrating-legacy-policies.mdx:94]:
```suggestion
keto relation-tuple parse alice_policies --format json | \
```

File: docs/keto/guides/migrating-legacy-policies.mdx [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+  keto relation-tuple create - >/dev/null \
+  && echo "Successfully created tuple" \
+  || echo "Encountered error"
+
+```
+
+Bob
+```bash
+keto relation-tuple parse bob_policies- --format json | \
```
Comment (zepatrik) [inline docs/keto/guides/migrating-legacy-policies.mdx:103]:
```suggestion
keto relation-tuple parse bob_policies --format json | \
```

File: docs/keto/guides/upgrade.mdx [hunk 11]
```diff
@@ -12,7 +12,9 @@ incorrectly.
 
 :::
 
-1. Review breaking changes.  
+1. Review breaking changes.
```
Comment (zepatrik) [inline docs/keto/guides/upgrade.mdx:15]: Is this a duplication?

**Review by zepatrik — CHANGES_REQUESTED** [review]
Nice, sounds like a good migration path. Maybe you could also add a small example for a check? I know that you link to other guides, but it probably helps to see a check in the context of this example.

File: docs/keto/guides/upgrade.mdx (hunk 11 above)
Comment (gen1us2k) [reply docs/keto/guides/upgrade.mdx:15]: Yep. Removed it. It was a merge issue

**Comment by gen1us2k** [comment]
@zepatrik I fixed comments and added check example

**Review by zepatrik — APPROVED** [review]

**Pull request merged**